lib/reform/result.rb

```diff
@@ -23,12 +23,16 @@ def add_error(key, error_text)
         CustomError.new(key, error_text, @results)
       end
 
+      def to_results
+        @results
+      end
+
       private
 
       # this doesn't do nested errors (e.g. )
       def filter_for(method, *args)
         @results.collect { |r| r.public_send(method, *args) }
-                .inject({}) { |hsh, errs| hsh.merge(errs) }
+                .inject({}) { |hah, err| hah.merge(err) { |key, old_v, new_v| (new_v.is_a?(Array) ? (old_v |= new_v) : old_v.merge(new_v)) } }
```

Comment:
this was overriding the values which are arrays...for example:
`{title: ["custom error"], title: ["must be filled"]}` -> `{title: ["must be filled"]}`